### PR TITLE
Add Navigation Drawer to edit name and change app to dark mode. 

### LINKED ISSUE #4: [FEATURE]: "Adding a Side Menu to the Maps interface"
**Describe the solution you'd like**
Need a side menu on the maps interface to act as a container for features such as _edit profile_ and the like as well as display the user's current profile info. Similar to Uber's side menu.

### PR DESCRIPTION
**Pull request by gazlaws-dev**
Closes #4

### CODE REVIEW

**Comment by rohit-s8** [comment]
rebase this branch onto testing as all new branches must have testing as their base as mentioned in the [README](https://github.com/EyyAuto/EyyAuto/blob/master/README.md)

**Comment by rohit-s8** [comment]
pressing the back button is closing the app while in ProfileActivity. I think the default action should be to revert to the previous activity without making any changes right?

**Comment by rohit-s8** [comment]
Also I see that you've added the side menu within ProfileActivity as well. So that provides us a way to go back to maps without using the back button. However, the back button should be the primary way for the user to return to maps deferring changes. Also it is debatable whether or not a side menu makes sense here.

**Comment by gazlaws-dev** [comment]
Yeah, I think I'll remove the drawer from that activity and only provide a back button on the action bar. That seems more natural.

**Comment by esah2111** [comment]
ive noticed something strange with the back button functionality.

consider this sccenario:
1--user registered as "Name1". this is the original name
2--name changed to "Name2" in the profile activity
3--name changed to "Name3" once again in the profile activity

now the db holds the latest name "Name3" and the app is in maps activity. the nav-drawer shows the name as "Name3"

4--hit back button
the name displayed in nav -drawer changes to "Name2"

5--hit back button again
the name displayed in nav -drawer changes to "Name1"

6--hit back button again
the app exits

the db however holds the latest update for the name ie "Name3"

**Comment by gazlaws-dev** [comment]
@esah2111 Regarding the hamburger menu, I'm not sure if the action bar will be needed after implementing the two search bars.  A lot of space will be wasted at the top of the screen. So I'll add that after the search bars are added and see how it looks

**Comment by rohit-s8** [comment]
> Good catch, it's because the old activities are still in the back stack. I'll change it so that instead of creating new activities on top, we'll just edit the name in the old one.

actually i see that you haven't called finish() after using startActivity in some places. This is the issue. finish() should generally be called after calling startActivity(). Adding it should fix the issue

File: app/src/main/java/in/ac/nitc/eyyauto/MapActivity.java [hunk 1]
```diff
@@ -67,19 +86,87 @@ public void onMapReady(GoogleMap googleMap) {
...
+
+
+    private void displayScreen(int itemId){
+        Log.d(TAG, "displayScreen: new Activity");
+        switch (itemId) {
+            case R.id.nav_profile:
+                Intent i = new Intent(MapActivity.this,ProfileActivity.class);
+                i.putExtra(INTENT_USER,user);
+                startActivityForResult(i, 1);
+                break;
```
Comment (rohit-s8) [inline app/src/main/java/in/ac/nitc/eyyauto/MapActivity.java]: As mentioned in the comments we need a finish() here after a startActivity(i)

**Comment by gazlaws-dev** [comment]
I was planning to implement on onActivityResult to update the name. The best practice for implementing back functionality is using startActivityForResult according to the Android developer docs. Plus in case the user has entered information in the search bars, saving the activity would be better UX.

File: app/src/main/java/in/ac/nitc/eyyauto/ProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            // Respond to the action bar's Up/Home button
+            case android.R.id.home:
+                Intent i = new Intent(ProfileActivity.this, MapActivity.class);
+                i.putExtra(INTENT_USER,user);
+                setResult(Activity.RESULT_OK,i);
```
Comment (rohit-s8) [inline app/src/main/java/in/ac/nitc/eyyauto/ProfileActivity.java]: It seems this is used in conjunction with startActivityForResult in MapActivity. But as mentioned in the comments this is useless without overriding onActivityResult. So unless you have a specific reason for getting a result from this activity it should not be used. Use startActivity instead.

**Comment by rohit-s8** [comment]
> I was planning to implement on onActivityResult to update the name. The best practice for implementing back functionality is using startActivityForResult according to the Android developer docs. Plus in case the user has entered information in the search bars, saving the activity would be better UX, rather than creating new activities every time

could you provide a link to the documentation you referenced?

For the issue you have mentioned, Isn't simpler and hence more fool proof to use onSaveInstanceState and onRestoreInstanceState. Managing Activities on the back stack could prove cumbersome and introduce unforeseen bugs

**Comment by gazlaws-dev** [comment]
I referenced this (https://developer.android.com/training/implementing-navigation/temporal) but I'll look into the link you posted. Yeah we'll keep it as simple as possible

**Comment by rohit-s8** [comment]
> I referenced this (https://developer.android.com/training/implementing-navigation/temporal) but I'll look into the link you posted. Yeah we'll keep it as simple as possible

I had a look at the link you sent but couldn't find anything on using startActivityForResult. Perhaps It's been updated since the last time you looked at it?

Either way, from what I understand, managing the back stack is mainly required when we want a custom back button flow, such as when using the navigation drawer to navigate to an Activity that should be preceded by other activities according to some fixed hierarchy in general. Or when using fragments in an Activity to determine what the UI should like.

Since our app won't be having complicated navigation flows nor is it using fragments I believe we can safely avoid this line of thought.

File: app/src/main/java/in/ac/nitc/eyyauto/ProfileActivity.java (hunk 2 above)
Comment (gazlaws-dev) [reply app/src/main/java/in/ac/nitc/eyyauto/ProfileActivity.java]: Will do. I'm a bit busy today so I'll fix it tomorrow.

**Comment by esah2111** [comment]
> > I referenced this (https://developer.android.com/training/implementing-navigation/temporal) but I'll look into the link you posted. Yeah we'll keep it as simple as possible
> 
> I had a look at the link you sent but couldn't find anything on using startActivityForResult. Perhaps It's been updated since the last time you looked at it?
> 
> Either way, from what I understand, managing the back stack is mainly required when we want a custom back button flow, such as when using the navigation drawer to navigate to an Activity that should be preceded by other activities according to some fixed hierarchy in general. Or when using fragments in an Activity to determine what the UI should like.
> 
> Since our app won't be having complicated navigation flows nor is it using fragments I believe we can safely avoid this line of thought.

we might be requiring a custom back button functionality once we implement the start ride button and also managing the start and destination address

even if we might not require that functionality in the future i think it would be better to have that beforehand so that laborious changes can be avoided in the future if needed

**Pull request merged**